### PR TITLE
Group api CRUD + tests

### PR DESCRIPTION
**Pull request by adrnchk**

### CODE REVIEW

File: DogeNet.BLL/Features/Group/CreateGroup/CreateGroupModelProfiles.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    using System.Threading.Tasks;
+    using AutoMapper;
+    using DogeNet.DAL.Models;
+
+    public class CreateGroupModelProfiles : Profile
+    {
+        public CreateGroupModelProfiles()
+        {
+            this.CreateMap<CreateGroupModel, Group>()
+                .ForMember(src => src.CreatedAt, opt => opt.MapFrom(c => DateTime.Now));
```
Comment (oleh-leobit) [inline DogeNet.BLL/Features/Group/CreateGroup/CreateGroupModelProfiles.cs:20]: Do we need to verify if image URLs are correct?

File: DogeNet.BLL/Features/Group/DeleteGroup/DeleteGroupHandler.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        private readonly DBContext context;
+
+        public DeleteGroupHandler(DBContext context)
+        {
+            this.context = context;
+        }
+
+        public Task<Unit> Handle(DeleteGroupCommand request, CancellationToken cancellationToken)
+        {
+            this.context.Groups.Remove(this.context.Groups.Find(request.id));
```
Comment (oleh-leobit) [inline DogeNet.BLL/Features/Group/DeleteGroup/DeleteGroupHandler.cs:23]: Why there is no validator for id?

File: DogeNet.BLL/Features/Group/EditGroup/EditGroupValidator.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using FluentValidation;
+
+    public class EditGroupValidator : AbstractValidator<EditGroupModel>
+    {
+        public EditGroupValidator()
+        {
+            this.RuleFor(enitity => enitity.Name).NotEmpty().NotNull().MinimumLength(GroupConstants.MinLengthForName)
```
Comment (oleh-leobit) [inline DogeNet.BLL/Features/Group/EditGroup/EditGroupValidator.cs:19]: Id validation?

File: DogeNet.BLL/Features/Group/GetGroup/GetGroupQueryValidator.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    using System.Text;
+    using System.Threading.Tasks;
+    using DogeNet.DAL;
+    using FluentValidation;
+
+    public class GetGroupQueryValidator : AbstractValidator<GetGroupQuery>
+    {
+        public GetGroupQueryValidator(DBContext db)
+        {
+            this.RuleFor(entity => entity.id).NotNull().Must(id => GroupChecks.PostIdValid(db, id)).WithMessage("Group not found");
```
Comment (oleh-leobit) [inline DogeNet.BLL/Features/Group/GetGroup/GetGroupQueryValidator.cs]: After merge please use extension here

File: DogeNet.BLL/Features/Group/GroupChecks.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using DogeNet.DAL;
+
+    public class GroupChecks
+    {
+        public static bool PostIdValid(DBContext db, int id) => db.Groups.Any(o => o.Id == id);
```
Comment (oleh-leobit) [inline DogeNet.BLL/Features/Group/GroupChecks.cs]: Should rename it: DoesGroupExist or something else

File: DogeNet.Tests/BLL/Features/Groups/CreateGroup/CreateGroupMappingProfileTests.cs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        public CreateGroupMappingProfileTests()
+        {
+            var config = new MapperConfiguration(cfg => cfg.AddProfile<CreateGroupModelProfiles>());
+            this.mapper = config.CreateMapper();
+        }
+
+
+        [Fact]
+        public void ToGroup()
+        {
```
Comment (oleh-leobit) [inline DogeNet.Tests/BLL/Features/Groups/CreateGroup/CreateGroupMappingProfileTests.cs:27]: Test should include error cases also

**Review by oleh-leobit — COMMENTED** [review]
There are a few comments that should be resolved

File: DogeNet.BLL/Features/Group/DeleteGroup/DeleteGroupHandler.cs (hunk 2 above)
Comment (adrnchk) [reply DogeNet.BLL/Features/Group/DeleteGroup/DeleteGroupHandler.cs:23]: done

File: DogeNet.BLL/Features/Group/EditGroup/EditGroupValidator.cs (hunk 3 above)
Comment (adrnchk) [reply DogeNet.BLL/Features/Group/EditGroup/EditGroupValidator.cs:19]: done

File: DogeNet.BLL/Features/Group/GetGroup/GetGroupQueryValidator.cs (hunk 4 above)
Comment (adrnchk) [reply DogeNet.BLL/Features/Group/GetGroup/GetGroupQueryValidator.cs]: ok, I'll create another pull request

File: DogeNet.BLL/Features/Group/GroupChecks.cs (hunk 5 above)
Comment (adrnchk) [reply DogeNet.BLL/Features/Group/GroupChecks.cs]: done

File: DogeNet.Tests/BLL/Features/Groups/CreateGroup/CreateGroupMappingProfileTests.cs (hunk 6 above)
Comment (adrnchk) [reply DogeNet.Tests/BLL/Features/Groups/CreateGroup/CreateGroupMappingProfileTests.cs:27]: done

File: DogeNet.BLL/Features/Group/CreateGroup/CreateGroupModelProfiles.cs (hunk 1 above)
Comment (adrnchk) [reply DogeNet.BLL/Features/Group/CreateGroup/CreateGroupModelProfiles.cs:20]: done, but I don't know in which format we will store images

File: DogeNet.BLL/Features/Group/CreateGroup/CreateGroupValidator.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    public class CreateGroupValidator : AbstractValidator<CreateGroupModel>
+    {
+        public CreateGroupValidator()
+        {
+            this.RuleFor(enitity => enitity.Name).NotEmpty().NotNull().MinimumLength(GroupConstants.MinLengthForName)
+                  .MaximumLength(GroupConstants.MaxLengthForName);
+            this.RuleFor(enitity => enitity.Title).NotNull().MaximumLength(GroupConstants.MaxLengthForTitle);
+            this.RuleFor(enitity => enitity.CreatorId).NotNull();
+            this.RuleFor(enitity => enitity.StatusId).NotNull();
+            this.RuleFor(entity => entity.CoverImg).Must(source => source == string.Empty || GroupChecks.IsUrlImgCorrect(source)).WithMessage("Invalid ImgUrl");
```
Comment (oleh-leobit) [inline DogeNet.BLL/Features/Group/CreateGroup/CreateGroupValidator.cs:23]: Move validation of empty string to IsUrlImgCorrect and make this an extension method

**Review by oleh-leobit — APPROVED** [review]
Please resolve comments and you are good to merge

File: DogeNet.BLL/Features/Group/CreateGroup/CreateGroupValidator.cs (hunk 7 above)
Comment (adrnchk) [reply DogeNet.BLL/Features/Group/CreateGroup/CreateGroupValidator.cs:23]: @oleh-leobit I want to make it possible to create group without avatar img

Comment (adrnchk) [reply DogeNet.BLL/Features/Group/CreateGroup/CreateGroupValidator.cs:23]: ok, I will make it as an extention in next pull request

**Pull request merged**